src/clients/victoriaMetrics/promAgent.ts

```diff
@@ -0,0 +1,92 @@
+import promClient from 'prom-client';
+import assert from 'assert';
+import { sendToDiscordAdmins } from '../discord';
+
+const VM_IMPORT_PROMETHEUS_URL =
+  'http://localhost:8428/api/v1/import/prometheus';
+
+const MAX_PUSH_METRICS_ERRORS = 5;
+const PUSH_METRICS_ERRORS_RATE_LIMIT = 60 * 60 * 1000; // 1 hour
+
+export class PromAgent {
+  private metricNames: Set<string>;
+  private pushMetricsErrorsTimestamps: number[] = [];
+
+  constructor() {
+    this.metricNames = new Set<string>();
+  }
+
+  public addMetricName(metricName: string) {
+    assert(
```

Comment:
Can you double check this completely takes down the server? And not just handled by the throw catch?

Reply:
It does not take down server, changing to process.exit